src/components/scenes/newAccount/NewAccountTosScene.tsx

```diff
@@ -210,3 +167,124 @@ const getStyles = cacheStyles((theme: Theme) => ({
     minHeight: theme.rem(6)
   }
 }))
+
+const setTouchOtp = async (account: EdgeAccount, dispatch: Dispatch) => {
+  await enableTouchId(account).catch(e => {
+    console.log(e) // Fail quietly
+  })
+  await account.dataStore.setItem(
+    Constants.OTP_REMINDER_STORE_NAME,
+    Constants.OTP_REMINDER_KEY_NAME_CREATED_AT,
+    Date.now().toString()
+  )
+  dispatch(loadTouchState())
+}
+
+/**
+ * Terms of Service scene for new regular or light accounts
+ */
+interface NewAccountTosProps extends SceneProps<'newAccountTos'> {
+  branding: Branding
+}
+export const NewAccountTosScene = (props: NewAccountTosProps) => {
+  const { route, branding } = props
+  const imports = useImports()
+  const { context } = imports
+  const dispatch = useDispatch()
+
+  const handleBack = useHandler((): void => {
+    dispatch({
+      type: 'NAVIGATE',
+      data: { name: 'newAccountPin', params: { ...route.params } }
+    })
+  })
+
+  const handleNext = useHandler(async () => {
+    logEvent(`Signup_Terms_Agree_and_Create_User`)
+    const { username, password, pin } = route.params
+
+    const account = await context.createAccount({
+      ...imports.accountOptions,
+      username,
+      password,
+      pin
+    })
+    account.watch('loggedIn', loggedIn => {
+      if (!loggedIn) dispatch({ type: 'RESET_APP' })
+    })
+    await setTouchOtp(account, dispatch)
+
+    dispatch({
+      type: 'NAVIGATE',
+      data: {
+        name: 'newAccountReview',
+        params: {
+          ...route.params,
+          account
+        }
+      }
+    })
+
+    logEvent('Signup_Create_User_Success')
+  })
+
+  return (
+    <TosComponent
+      hidePasswordTerms={route.params.username == null}
+      branding={branding}
+      onBack={handleBack}
+      onNext={handleNext}
+    />
+  )
+}
+
+/**
+ * Terms of Service scene for upgrading existing (light) accounts
+ */
+interface UpgradeTosProps extends SceneProps<'upgradeTos'> {
+  branding: Branding
+}
+export const UpgradeTosScene = (props: UpgradeTosProps) => {
+  const { route, branding } = props
+  const dispatch = useDispatch()
+
+  const handleBack = useHandler((): void => {
+    dispatch({
+      type: 'NAVIGATE',
+      data: { name: 'upgradeUsername', params: { ...route.params } }
+    })
+  })
+
+  const handleNext = useHandler(async () => {
+    logEvent(`Signup_Terms_Agree_and_Back_Up_User`)
+    const { account, username, password } = route.params
+
+    if (username == null || password == null)
+      throw new Error('Failed to update account, missing username or password')
+    await account.changeUsername({
+      username,
+      password
+    })
+    await setTouchOtp(account, dispatch)
```

Comment:
I don't think we need to do `setTouchOtp` here, because it was already done when the account was created. Unless... we want to reset the date at this point in time? In that case, we just need to stomp the date, but not activate touch, since that's still done at create time.